src/electionguard/group.cpp

```diff
@@ -755,38 +755,32 @@ namespace electionguard
     unique_ptr<ElementModQ> a_plus_bc_mod_q(const ElementModQ &a, const ElementModQ &b,
                                             const ElementModQ &c)
     {
-        // since the result of b*c can be larger than Q's 256,
-        // use P's length and 4096 Hacl for rest of calculation
+        // multiply b * c and the result will be twice Q in size
+        uint64_t bc[MAX_Q_LEN * 2] = {};
+        Hacl_Bignum256_mul(const_cast<ElementModQ &>(b).get(), const_cast<ElementModQ &>(c).get(),
```

Comment:
Is there any reason that Bignum256::mul, Bignum256::mod, and Bignum256::add were not used since they ultimately call the same thing under the hood?

Reply:
Thanks Steve, changed it to use the abstraction layer.